caikit/interfaces/runtime/data_model/info.py

```diff
@@ -0,0 +1,40 @@
+# Copyright The Caikit Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+This file contains interfaces to handle information requests
+"""
+
+# Standard
+from typing import Dict, Optional
+
+# First Party
+import alog
+
+# Local
+from caikit.core.data_model import PACKAGE_COMMON, DataObjectBase, dataobject
+
+log = alog.use_channel("RUNTIMEOPS")
+
+RUNTIME_PACKAGE = f"{PACKAGE_COMMON}.runtime"
+
+
+@dataobject(RUNTIME_PACKAGE)
+class RuntimeInfoRequest(DataObjectBase):
+    pass
+
+
+@dataobject(RUNTIME_PACKAGE)
+class RuntimeInfoResponse(DataObjectBase):
+    runtime_version: Optional[str]
+    python_packages: Dict[str, str]
```

Comment:
Crap, One more thing: We should explicitly set `FieldNumber`s on these:

```py
from py_to_proto.dataclass_to_proto import Annotated, FieldNumber
...

class RuntimeInfoResponse(DataObjectBase):
    runtime_version: Annotated[Optional[str], FieldNumber(1)]
    python_packages: Annotated[Dict[str, str], FieldNumber(2)]
```

Reply:
I noticed the `Annotated` on other data models and was wondering, I'm guessing these should be set for all responses?

Reply:
yep, those would need to be for all responses.

Reply:
It's not strictly required, but it guarantees that the messages will not accidentally change their proto representations by changing field orders. We require that all messages in `caikit` itself have them.